File: code/src/database/DatabaseSelector.py
from datetime import date,datetime
import json

def DatabaseName(databaseDate,provider):
    databaseName =''
    if(provider == 'IITM' or provider == 'ECMWF'):
        today = datetime.strptime(databaseDate,"%Y%m%d")
    if(provider == 'ACTUAL'):
        today = datetime.strptime(databaseDate,"%d_%m_%Y")
   
    todayDateFormat = int(today.strftime("%Y%m%d"))
    
    with open('D:\\Software\\Forecasting DB\\DatabaseRecord.json') as f:
        dbRecord = json.load(f)
        timeSeriesArray = dbRecord.keys()
        for series in timeSeriesArray:
            datesData = series.split('-')
            dbStartDate = int(datesData[0])
            dbEndDate = int(datesData[1])
            if(dbStartDate <= todayDateFormat and todayDateFormat <= dbEndDate):
                databaseName = dbRecord[series]
    return databaseName

File: code/src/database/Database-Selector.js
function DatabaseName(date){
    var databaseName ;
    var todayDate;
    
    todayDate =new Date(date);
    
    
    var todayDateFormat = parseInt(todayDate.getFullYear() + '' + ('0' + (todayDate.getMonth() + 1)).slice(-2) + "" + ('0' + todayDate.getDate()).slice(-2));
    var dbRecord = JSON.parse(require('fs').readFileSync('D:\\Software\\Forecasting DB\\DatabaseRecord.json'))
    var timeSeriesArray = Object.keys(dbRecord)
    timeSeriesArray.forEach( (series) => {
        datesData = series.split('-')
        dbStartDate = parseInt(datesData[0])
        dbEndDate = parseInt(datesData[1])
        
        if(dbStartDate <= todayDateFormat && todayDateFormat <= dbEndDate){
            databaseName = dbRecord[series]
        }
    })
    return databaseName;
}

// var giveDate = new Date('2020-01-10')
// console.log('The everydataind si ',giveDate.getDate(),' ',giveDate.getMonth())
// console.log('THe dataabase returned is ',DatabaseName(giveDate))
module.exports.DatabaseName = DatabaseName;